matchzoo/models/arci_model.py

```diff
@@ -52,12 +50,28 @@ def _conv_pool_block(self, input: typing.Any, kernel_count: int,
         output = MaxPooling1D(pool_size=pool_size)(output)
         return output
 
+    def set_embedding_mat(self, embedding_mat: np.ndarray):
+        """
+        Set pretrained embedding for ArcI model.
+
+        :param embedding_mat: pretrained embedding in numpy format.
+        """
+        self._params['embedding_mat'] = embedding_mat
+        self._params['vocab_size'], self._params['embedding_dim'] = \
+            embedding_mat.shape
+
     def build(self):
         """
         Build model structure.
 
         ArcI use Siamese arthitecture.
         """
+        # Check if provided embedding matrix
+        if self._params['embedding_mat'] is None:
+            self._params['embedding_mat'] = \
+               np.random.uniform(-0.2, 0.2, (self._params['vocab_size'],
+                                             self._params['embedding_dim']))
```

Comment:
this shouldn't exist in the model, you can use getter here to get embedding matrix, if none raise exception. And manually set embedding matrix in unit test and integration test.